library/public/engine.h

```diff
@@ -0,0 +1,142 @@
+/**
+ * @class   engine
+ * @brief   Class used to create instance of other classes
+ *
+ * The main class of the libf3d to create all other instances
+ * Configured on creation using an enum, then all objects
+ * can be accessed through their getter.
+ *
+ * Example usage:
+ *
+ * \code{.cpp}
+ *  f3d::engine eng(f3d::engine::CREATE_WINDOW | f3d::engine::CREATE_INTERACTOR);
+ *  f3d::loader& load = eng.getLoader();
+ *  load.addFile("path/to/file");
+ *  load.loadFile(f3d::loader::LoadFileEnum::LOAD_CURRENT);
+ *  f3d::interactor& inter = eng.getInteractor();
+ *  inter.start();
+ * \endcode
+ */
+
+#ifndef f3d_engine_h
+#define f3d_engine_h
+
+#include "export.h"
+#include "window.h"
+
+#include <map>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace f3d
+{
+class options;
```

Comment:
You should consider including these classes instead of forward declaring them. There is some debate on forward declarations vs includes, but it increases usability of headers when they are self contained.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf11-header-files-should-be-self-contained

Now, something like `engine::internals` Pimpl declaration should definitely be forward declared, so I'm not saying don't ever forward declare, but consider preferring including thing that a user needs to use the class.

Reply:
@Meakk https://stackoverflow.com/questions/9906402/should-one-use-forward-declarations-instead-of-includes-wherever-possible. I think this way of fwd declaration is idomatic from VTK, but do not make sens in the context of the public header of the libf3d.

Reply:
Adressed in https://github.com/f3d-app/f3d/pull/355